src/Operations.jl

```diff
@@ -46,8 +46,21 @@ end
 is_dep(ctx::Context, pkg::PackageSpec) =
     any(uuid -> uuid == pkg.uuid, [uuid for (name, uuid) in ctx.env.project.deps])
 
-function load_direct_deps!(ctx::Context, pkgs::Vector{PackageSpec}; version::Bool=true)
-    # load rest of deps normally
+#TODO rename
+function load_version(version, fixed, preserve::PreserveLevel)
+    version === nothing && return VersionSpec() # for stdlibs
+    fixed && return version
+    preserve == PRESERVE_ALL && return something(version, VersionSpec())
```

Comment:
Maybe clearer as an `if-ifelse-`block

Reply:
yea, looks pretty ugly like how it is now :laughing: